index.d.ts

```diff
@@ -18,13 +18,14 @@ type BaseStdioOption =
 	| 'ignore'
 	| 'inherit';
 
-type CommonStdioOption =
+type CommonStdioOption<IsSync extends boolean = boolean> =
 	| BaseStdioOption
 	| 'ipc'
 	| number
 	| undefined
 	| URL
-	| {file: string};
+	| {file: string}
+	| IfAsync<IsSync, ((chunks: Iterable<string | Uint8Array>) => AsyncGenerator<string | Uint8Array, void, void>)>;
```

Comment:
Technically, `chunks` is `Iterable<Uint8Array>` if the `encoding` option is `buffer`, and `Iterable<string>`.

It turns out typing this was quite difficult, so I left it with this weaker typing for the time being.

Reply:
Can you add a TODO comment so it won't be forgotten?

Reply:
Done in b760d9efdf2748e70683102028e5bbace66ce9e8 and also as a new issue in https://github.com/sindresorhus/execa/issues/694